Source/Details/ASPageTable.m

```diff
@@ -148,4 +148,17 @@ - (void)removeObjectForPage:(ASPageCoordinate)page
   [self removeObjectForKey:key];
 }
 
+- (NSMapTable *)deepCopy
+{
+  NSMapTable *result = [[NSMapTable alloc] initWithKeyPointerFunctions:self.keyPointerFunctions
+                                                 valuePointerFunctions:self.valuePointerFunctions
+                                                              capacity:self.count];
+
+  for (id key in self) {
+    [result setObject:[[self objectForKey:key] copy] forKey:key];
+  }
```

Comment:
Valid questions!

I ended up removing this method after implementing a method that walks through the table and gathers unmeasured layout attributes, which is what I should have done in the first place.